### PR TITLE
feat(python): add freeze_panes option to write_excel

### LINKED ISSUE #9961: Freeze panes
### Problem description

Can a "freeze_panes" argument be added the write_excel function? This is an xlsxwriter option so I wonder if it would be difficult to add?

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #9961

### CODE REVIEW

**Review by alexander-beedie — CHANGES_REQUESTED** [review]
Parameter typing and associated docstring should account for "A1" notation support (the first two param values can be replaced with a single string of that form). Also, even though we can't verify that it has had an effect, do include it in one of the unit tests so that we at least know it doesn't blow up :)

File: py-polars/polars/dataframe/frame.py [hunk 1]
```diff
@@ -2980,6 +2993,10 @@ def write_excel(
                 )
             ws.autofit()
 
+        # freeze panes
```
Comment (alexander-beedie) [inline py-polars/polars/dataframe/frame.py:2996]: Don't really need a "#freeze panes" comment above `if freeze_panes:`  :)

File: py-polars/polars/dataframe/frame.py [hunk 2]
```diff
@@ -2549,6 +2549,7 @@ def write_excel(
         hidden_columns: Sequence[str] | None = None,
         hide_gridlines: bool = False,
         sheet_zoom: int | None = None,
+        freeze_panes: str | tuple[int, int] | tuple[int, int, int, int] | None = None,
```
Comment (alexander-beedie) [inline py-polars/polars/dataframe/frame.py:2552]: Could also be `[str, int, int]` ;)

Comment (mcrumiller) [reply py-polars/polars/dataframe/frame.py:2552]: Can it? It's not in their documentation, granted they don't really have a well-defined API: https://xlsxwriter.readthedocs.io/worksheet.html#freeze_panes. I suppose a simple test will tell. I have a meeting but I can do this in an hour.

Comment (alexander-beedie) [reply py-polars/polars/dataframe/frame.py]: Yup; it's implicit in the first two parameter values being replaceable with "A1" notation - that remains true even if then add the optional final two.

**Comment by mcrumiller** [comment]
@alexander-beedie I think I've hit all of your concerns.

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**